src/main/java/uk/gov/cshr/civilservant/controller/AgencyTokenController.java

```diff
@@ -0,0 +1,90 @@
+        this.organisationalUnitService = organisationalUnitService;
+    }
+
+    @GetMapping
+    public ResponseEntity getAgencyTokens() {
+        return ResponseEntity.ok(agencyTokenService.getAllAgencyTokens());
+    }
+
+    @GetMapping(params = {"domain"})
+    public ResponseEntity getAgencyTokensByDomain(@RequestParam String domain) {
+        return ResponseEntity.ok(agencyTokenService.getAllAgencyTokensByDomain(domain));
+    }
+
+    @GetMapping(params = {"domain", "token"})
+    public ResponseEntity getAgencyTokensByDomainAndToken(@RequestParam String domain, @RequestParam String token) {
+        return agencyTokenService.getAgencyTokenByDomainAndToken(domain, token)
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @GetMapping(params = {"domain", "code"})
+    public ResponseEntity getAgencyTokensByDomainAndOrganisationalUnit(@RequestParam String domain, @RequestParam String code) {
+        return agencyTokenService.getAgencyTokenByDomainAndOrganisation(domain, code)
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @GetMapping(params = {"domain", "token", "code"})
+    public ResponseEntity getAgencyTokensByDomainTokenAndOrganisationalUnit(@RequestParam String domain, @RequestParam String token, @RequestParam String code) {
+        return agencyTokenService.getAgencyTokenByDomainTokenAndOrganisation(domain, token, code)
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @PutMapping
+    public ResponseEntity updateSpaceAvailable(@RequestBody UpdateSpacesForAgencyTokenRequestDTO updateSpacesForAgencyTokenRequestDTO) {
+        try {
+            log.info("Updating agency token with parameters domain=" + updateSpacesForAgencyTokenRequestDTO.getDomain() +
+                    " token=" + updateSpacesForAgencyTokenRequestDTO.getToken() +
+                    " code=" + updateSpacesForAgencyTokenRequestDTO.getCode() +
+                    " isRemoveUser=" + updateSpacesForAgencyTokenRequestDTO.isRemoveUser());
+
+            List<OrganisationalUnit> passedOrganisationalUnitList = new ArrayList<>();
+            List<String> organisationalUnitCodeList = new ArrayList<>();
+            passedOrganisationalUnitList = organisationalUnitService.getOrganisationWithParents(updateSpacesForAgencyTokenRequestDTO.getCode());
+            for (OrganisationalUnit organisationalUnit : passedOrganisationalUnitList) {
+                organisationalUnitCodeList.add(organisationalUnit.getCode());
+            }
+            agencyTokenService.updateAgencyTokenSpacesAvailable(updateSpacesForAgencyTokenRequestDTO.getDomain(), updateSpacesForAgencyTokenRequestDTO.getToken(), organisationalUnitCodeList, updateSpacesForAgencyTokenRequestDTO.isRemoveUser());
```

Comment:
Could we replace lines 69-74 with something like the following:

```
List<String> organisationalUnitCodeList = organisationalUnitService.getOrganisationWithParents(updateSpacesForAgencyTokenRequestDTO.getCode())
                    .stream()
                    .map(OrganisationalUnit::getCode)
                    .collect(Collectors.toList());
```

Should tidy up and reduce number of vars being created

Reply:
Updated.